File: Julia/algorithmic_toolbox/04-fibonacci-last-digit.jl
const modulus = 10


function fibonaccilastdigit(n)
    k = 1
    a, b = 0, 1
    while k <= n
        a, b = b, (a + b) % modulus
        k += 1
    end
    a
end


function main()
    n = parse(Int, readline())
    digit = fibonaccilastdigit(n)
    println(digit)
end


main()

File: Rust/algorithmic_toolbox/src/bin/04_fibonacci_last_digit.rs
use std::io;

const MODULUS: i32 = 10;


fn read_int() -> i32 {
    let mut x = String::new();
    io::stdin().read_line(&mut x).unwrap();
    x.trim().parse().unwrap()
}

fn fibonacci_last_digit(n: i32) -> i32 {
    let mut k = 1;
    let mut a = 0;
    let mut b = 1;
    while k <= n {
        (a, b) = (b, (a + b) % MODULUS);
        k += 1;
    }
    a
}

fn main() {
    let n = read_int();
    let result = fibonacci_last_digit(n);
    println!("{}", result);
}
